pkg/cache/cache.go

```diff
@@ -166,8 +166,38 @@ type Options struct {
 	//
 	// This is a global setting for all objects, and can be overridden by the ByObject setting.
 	UnsafeDisableDeepCopy *bool
+
+	// MissPolicy determines how the cache should behave when no informers are started for the
+	// resource type a client is asking for in a Get or a List. See the CacheMissPolicies for
+	// documentation for each policy.
+	MissPolicy CacheMissPolicy
 }
 
+type CacheMissPolicy string
+
+const (
+	// CacheMissPolicyFail configures the cache to return a ErrResourceNotCached error when a user
+	// requests a resource the cache is not configured to hold. This error is distinct from an
+	// errors.NotFound.
+	CacheMissPolicyFail CacheMissPolicy = "fail"
```

Comment:
Can we call this `Strict` and call out in the comment that's the default?

Reply:
The failure explained here is `ErrResourceNotCached`, which is a soft error given we can always do a live lookup manually from a user side, or rework the cache to cache the new resource. IMO a Strict mode signals that the cache will never return something that it's not configured to hold and we can expand this mode later if we want to. The word Fail makes me think about a terminal failure, or panic, which isn't the case.

@sbueringer @alvaroaleman ?

Reply:
As far as the cache is concerned, it does fail. Up to the clients to do what they want with it. I don't have a strong opinion, as long as the options read like English and are parallel to each other. `"fail"` and `"backfill"` are verbs and parallel. What would you like to pair with `"strict"`?